pkg/maintainer/spv/spv.go

```diff
@@ -57,10 +105,348 @@ func (sm *spvMaintainer) startControlLoop(ctx context.Context) {
+		if err != nil {
+			return fmt.Errorf(
+				"failed to get Bitcoin epoch difficulties: [%v]",
+				err,
+			)
+		}
+
+		if firstBlockHeaderDifficulty.Cmp(currentDifficulty) != 0 &&
+			firstBlockHeaderDifficulty.Cmp(previousDifficulty) != 0 {
+			// Skip the transaction as the difficulty of the first block
+			// header in the proof does not match the current or the
+			// previous difficulty seen on-chain. The reason for this is
+			// most likely that the proof consists of block headers from the
+			// beginning of a new Bitcoin difficulty epoch and the epoch is
+			// not proven on-chain yet. In that case the transaction will be
+			// proven in the future. The other reason could be that the
+			// transaction is older than the last two Bitcoin difficulty
+			// epoch. In that case the transaction will soon leave the
+			// sliding window of recent transactions.
+			continue
+		}
+
+		mainUTXO, vault, err := parseTransactionInputs(
+			sm.btcChain,
+			sm.spvChain,
+			transaction,
+		)
+		if err != nil {
+			return fmt.Errorf(
+				"error while parsing transaction inputs: [%v]",
+				err,
+			)
+		}
+
+		if err := sm.spvChain.SubmitDepositSweepProofWithReimbursement(
+			transaction,
+			proof,
+			mainUTXO,
+			vault,
+		); err != nil {
+			return fmt.Errorf(
+				"failed to submit deposit sweep proof with reimbursement: [%v]",
+				err,
+			)
+		}
+	}
 
 	return nil
 }
+
+func (sm *spvMaintainer) getUnprovenDepositSweepTransactions() (
+	[]*bitcoin.Transaction,
+	error,
+) {
+	blockCounter, err := sm.spvChain.BlockCounter()
+	if err != nil {
+		return nil, fmt.Errorf("failed to get block counter: [%v]", err)
+	}
+
+	currentBlock, err := blockCounter.CurrentBlock()
+	if err != nil {
+		return nil, fmt.Errorf("failed to get current block: [%v]", err)
+	}
+
+	// Calculate the starting block of the range in which the events will be
+	// searched for.
+	startBlock := currentBlock - sm.config.HistoryDepth
+
+	depositSweepTransactionProposals, err :=
+		sm.spvChain.PastDepositSweepProposalSubmittedEvents(
+			&tbtc.DepositSweepProposalSubmittedEventFilter{
+				StartBlock: startBlock,
+			},
+		)
+	if err != nil {
+		return nil, fmt.Errorf(
+			"failed to get past deposit sweep proposal submitted events: [%v]",
+			err,
+		)
+	}
+
+	// There will often be multiple events emitted for a single wallet. Prepare
+	// a list of unique wallet public key hashes.
+	walletPublicKeyHashes := uniqueWalletPublicKeyHashes(
+		depositSweepTransactionProposals,
+	)
+
+	unprovenDepositSweepTransactions := []*bitcoin.Transaction{}
+
+	for _, walletPublicKeyHash := range walletPublicKeyHashes {
+		wallet, err := sm.spvChain.GetWallet(walletPublicKeyHash)
+		if err != nil {
+			return nil, fmt.Errorf("failed to get wallet: [%v]", err)
+		}
+
+		if wallet.State != tbtc.StateLive &&
+			wallet.State != tbtc.StateMovingFunds {
+			// The wallet can only submit deposit sweep proofs if it's `Live` or
+			// `MovingFunds`. If the state is different skip it.
+			continue
```

Comment:
Let's add an info-level log for this case.

Reply:
Done in 132ac7d666c8964b47a71255b3c50334f240fe22.